drivers/peci/peci_handlers.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/peci.h>
+#include <syscall_handler.h>
+
+static inline int z_vrfy_peci_config(struct device *dev, u32_t bitrate)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, config));
+
+	return z_impl_peci_config((struct device *)dev, (u16_t)bitrate);
+}
+#include <syscalls/peci_config_mrsh.c>
+
+static inline int z_vrfy_peci_enable(struct device *dev)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, enable));
+
+	return z_impl_peci_enable((struct device *)dev);
+}
+#include <syscalls/peci_enable_mrsh.c>
+
+static inline int z_vrfy_peci_disable(struct device *dev)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, disable));
+
+	return z_impl_peci_disable((struct device *)dev);
+}
+#include <syscalls/peci_disable_mrsh.c>
+
+static inline int z_vrfy_peci_transfer(struct device *dev,
+				       const struct peci_msg *msg)
+{
+	struct peci_msg msg_copy;
+	struct peci_buf tx_buf_copy;
+	struct peci_buf rx_buf_copy;
+	const struct peci_buf *tx_buf = msg->tx_buffer;
+	const struct peci_buf *rx_buf = msg->rx_buffer;
+
+	Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, transfer));
+
+	/* Use a local copy to avoid switcheroo attacks. */
+	Z_OOPS(Z_SYSCALL_MEMORY_READ((const struct peci_msg *)msg,
```

Comment:
the z_user_from_copy() implicitly does the Z_SYSCALL_MEMORY_READ on the source pointer, delete this

Reply:
Removing peci_handlers for now. To be added in separate PR